From b41ac28947acc433247ff6db17d760b8a6323aa1 Mon Sep 17 00:00:00 2001
From: Ping Gao <ping.a.gao@intel.com>
Date: Wed, 6 Sep 2017 19:29:48 +0800
Subject: [PATCH 0926/1132] drm/i915/gvt: Enable guest conformance detection

The first overall comparison happen before the wordload submission.

Signed-off-by: Ping Gao <ping.a.gao@intel.com>
Reviewed-by: Kevin Tian
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>
Reviewed-on:
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>

V2: rebase 4.19
Signed-off-by: Xinyun Liu <xinyun.liu@intel.com>
---
 drivers/gpu/drm/i915/gvt/mmio_context.h | 1 +
 drivers/gpu/drm/i915/gvt/scheduler.c    | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/gvt/mmio_context.h b/drivers/gpu/drm/i915/gvt/mmio_context.h
index 5c3b9ff..b5059de 100644
--- a/drivers/gpu/drm/i915/gvt/mmio_context.h
+++ b/drivers/gpu/drm/i915/gvt/mmio_context.h
@@ -54,4 +54,5 @@ bool is_inhibit_context(struct intel_context *ce);
 int intel_vgpu_restore_inhibit_context(struct intel_vgpu *vgpu,
 				       struct i915_request *req);
 
+int intel_gvt_vgpu_conformance_check(struct intel_vgpu *vgpu, int ring_id);
 #endif
diff --git a/drivers/gpu/drm/i915/gvt/scheduler.c b/drivers/gpu/drm/i915/gvt/scheduler.c
index ebba07a..6dde102 100644
--- a/drivers/gpu/drm/i915/gvt/scheduler.c
+++ b/drivers/gpu/drm/i915/gvt/scheduler.c
@@ -711,6 +711,10 @@ static int dispatch_workload(struct intel_vgpu_workload *workload)
 	mutex_lock(&dev_priv->drm.struct_mutex);
 
 	ret = intel_gvt_scan_and_shadow_workload(workload);
+
+	if (intel_gvt_vgpu_conformance_check(vgpu, ring_id))
+		gvt_err("vgpu%d unconformance guest detected\n", vgpu->id);
+
 	if (ret)
 		goto out;
 
-- 
2.7.4

